docs/deployment/uwsgi_cookiecutter_1__nginx.rst

```diff
@@ -0,0 +1,294 @@
+    socket.  However, you'll need to make sure that *two* users have access to
+    change into the ``~/myproject/tmp`` directory: your current user (mine is
+    ``ubuntu`` and the user that Nginx will run as often named ``www-data`` or
+    ``nginx``).
+
+#.  Add a ``[uwsgi]`` section to ``production.ini``. Here are the lines
+    to include:
+
+    .. code-block:: text
+
+        [uwsgi]
+        proj = myproject
+        chdir = /home/ubuntu/%(proj)
+        processes = 2
+        threads = 2
+        offload-threads = 2
+        stats =  127.0.0.1:9191
+        max-requests = 5000
+        master = True
+        vacuum = True
+        enable-threads = true
+        harakiri = 60
+        chmod-socket = 020
+        plugin = python3
+        pidfile=%(chdir)/tmp/%(proj).pid
+        socket = %(chdir)/tmp/%(proj).sock
+        virtualenv = %(chdir)/env
+        uid = ubuntu
+        gid = www-data
+        #wsgi-file = wsgi.py                 # Used in part 2 of this tutorial
+        #callable = app                      # Used in part 2 of this tutorial
+        #logto = /var/log/uwsgi/%(proj).log  # Used in part 2 of this tutorial
+
+    And here is an explanation of the salient options:
+
+    .. code-block:: text
+
+        # Explanation of Options
+        #
+        # proj = myproject                    # Set a variable named "proj"
+        #                                       so we can use it elsewhere in this
+        #                                       block of config
+        #
+        # chmod-socket = 020                  # Change permissions on socket to
+        #                                       at least 020 so that in combination
+        #                                       with "--gid www-data", Nginx will be able
+        #                                       to write to it after  uWSGI creates it
+        #
+        # enable-threads                      # Execute threads that are in your app
+        #
+        # plugin = python3                    # Use the python3 plugin
+        #
+        # socket = %(chdir)/tmp/%(proj).sock  # Where to put the unix socket
+        # pidfile=%(chdir)/tmp/%(proj).pid    # Where to put PID file
+        #
+        # uid = ubuntu                        # Masquerade as the ubuntu user
+        #                                       This grants you permissions to use
+        #                                       python packages installed in your
+        #                                       home directory
+        #
+        # gid = www-data                      # Masquerade as the www-data group
+        #                                       This makes it easy to allow Nginx
+        #                                       (which runs as the www-data group)
+        #                                       access to the socket file.
+        #
+        # virtualenv = (chdir)/env            # Use packages installed in your venv
```

Comment:
"venv" is a tool to create virtual environments. Please use the correct term "virtual environment".